unified-codes/tools/scripts/dgraph/migration/v2/DataLoader.ts

```diff
@@ -1,118 +1,163 @@
-                } catch {
-                    console.log(`WARNING: Failed to load child with code ${child.code} for entity with code ${entity.code}`);
-                } finally {
-                    txn.discard();
-                }
-            }
+        req.setQuery(query);
+        req.setMutationsList([mutation]);
+        req.setCommitNow(true);
+
+        const txn = dgraphClient.newTxn();
+        try {
+          await txn.doRequest(req);
+          console.log(
+            `INFO: Loaded child with code ${child.code} for entity with code ${entity.code}`
+          );
+        } catch {
+          console.log(
+            `WARNING: Failed to load child with code ${child.code} for entity with code ${entity.code}`
+          );
+        } finally {
+          txn.discard();
         }
+      }
+    }
+
+    // Link properties to entities.
+    for await (const entity of entities.filter((entity) => !!entity.properties)) {
+      for await (const property of entity.properties) {
+        const req = new dgraph.Request();
+        const query = `query {
+                      Entity as var(func: eq(dgraph.type, ${entity.type})) @filter(eq(code, ${entity.code}))
+                      Property as var(func: eq(dgraph.type, ${property.type}))
+                  }`;
+
+        const mutation = new dgraph.Mutation();
+        mutation.setSetNquads(`
+            uid(Property) <value> "${property.value}" .
+            uid(Property) <dgraph.type> "${property.type}" .
+
+            uid(Entity) <properties> uid(Property) .
+        `);
+
+        req.setQuery(query);
+        req.setMutationsList([mutation]);
+        req.setCommitNow(true);
+
+        const txn = dgraphClient.newTxn();
+        try {
+          await txn.doRequest(req);
+          console.log(
+            `INFO: Loaded property with type ${property.type} for entity with code ${entity.code}`
+          );
+        } catch {
+          console.log(
+            `WARNING: Failed to load property with type ${property.type} for entity with code ${entity.code}`
+          );
+        } finally {
+          txn.discard();
+        }
+      }
+    }
 
-        // Link products to combinations.
-        for await (const entity of entities.filter(entity => !!entity.combines)) {
-            for await (const sibling of entity.combines) {
-                const req = new dgraph.Request();
-                const query = `query {
+    // Link products to combinations.
+    for await (const entity of entities.filter((entity) => !!entity.combines)) {
+      for await (const sibling of entity.combines) {
+        const req = new dgraph.Request();
+        const query = `query {
                     Entity as var(func: eq(dgraph.type, ${entity.type})) @filter(eq(code, ${entity.code}))
                     Sibling as var(func: eq(dgraph.type, ${sibling.type})) @filter(eq(code, ${sibling.code}))
```

Comment:
Thanks @katherine-sussol!
The rejected query has been fixed: 81613de06
Main problem is that the codes are now all lowercase, and `eq` is case sensitive. I've transformed in the query, and handled the no results scenario, so that we can return a null result.

Reply:
The combinations: another instance of an original bug. 
there was this: `this.graph[uc2].combines.push({ code: uc });`
and then 
```
for await (const sibling of entity.combines) {
...
Sibling as var(func: eq(dgraph.type, ${sibling.type})) @filter(eq(code, ${sibling.code}))```

given the way loader is working.. there is no `type` on `sibling` so new entities are created instead of looking up the correct ones.

Reply:
Fixed in 1a018398e ( other branch )